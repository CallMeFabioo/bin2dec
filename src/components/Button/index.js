import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Button = ({ disabled, children, ...props }) => (
	<button
		className={classnames({
			border: true,
			rounded: true,
			disabled,
			'px-4': true,
			'py-2': true,
			'font-semibold': true,
			'text-blue-700': true,
			'bg-transparent': true,
			'border-blue-500': true,
			'hover:bg-blue-500': true,
			'hover:text-white': true,
			'hover:border-transparent': true,
		})}
		{...props}
	>
		{children}
	</button>
);

Button.displayName = 'Button';

Button.propTypes = {
	disabled: PropTypes.bool,
};

export default Button;
