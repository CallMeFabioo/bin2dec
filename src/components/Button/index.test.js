import React from 'react';
import { render, screen } from '@testing-library/react';

import Button from '.';

it('renders a button', () => {
	const { container } = render(<Button>Button</Button>);

	expect(container).toMatchSnapshot();
});

it('renders a enabled button', () => {
	render(<Button>Button</Button>);

	expect(screen.getByRole('button')).not.toHaveClass('disabled');
});

it('renders a disabled button', () => {
	render(<Button disabled={true}>Button</Button>);

	expect(screen.getByRole('button')).toHaveClass('disabled');
});
