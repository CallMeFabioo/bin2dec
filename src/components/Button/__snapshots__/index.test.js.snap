// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button 1`] = `
<div>
  <button
    class="border rounded px-4 py-2 font-semibold text-blue-700 bg-transparent border-blue-500 hover:bg-blue-500 hover:text-white hover:border-transparent"
  >
    Button
  </button>
</div>
`;
